src/Apps/Checkout/ApplePay.tsx

```diff
@@ -0,0 +1,49 @@
+import { ApplePay_order$key } from "__generated__/ApplePay_order.graphql"
+import { ExpressCheckoutQueryRenderer } from "Apps/Order/Components/ExpressCheckout"
+import { graphql, useFragment } from "react-relay"
+import { useFeatureFlag } from "System/Hooks/useFeatureFlag"
+import { extractNodes } from "Utils/extractNodes"
+
+interface ApplePayProps {
+  order: ApplePay_order$key
+}
+
+export const ApplePay: React.FC<ApplePayProps> = ({ order }) => {
+  const data = useFragment(orderFragment, order)
+  const expressCheckoutPrototypeEnabled = useFeatureFlag(
+    "emerald_stripe-express-checkout-prototype",
+  )
+
+  if (!data) {
+    return null
+  }
+
+  const isOffer = data.mode === "OFFER"
+  const isFixedShipping = !!extractNodes(data.lineItems)[0]?.artwork
+    ?.isFixedShippingFeeOnly
+
+  const isExpressCheckoutEligible =
+    expressCheckoutPrototypeEnabled && !isOffer && isFixedShipping
+
+  if (!isExpressCheckoutEligible) {
+    return null
+  }
```

Comment:
This logic about availability for apple pay is the same as this: https://github.com/artsy/force/blob/24c32c2d02c772ca01e96485e48e0c26fc5d4a64/src/Apps/Order/Routes/Shipping/index.tsx#L81 Wonder if we should try to extract it somewhere?

Reply:
I don't think we should extract it now, since it's used in the ShippingContext, which I'm assuming that we might use when we do the redesign
I just added it here to match the current criteria we have to render the Express Checkout button :)

but maybe I can add a comment to refactor it when we work on the redesign

Reply:
I wondered if it should all go _inside_ the ExpressCheckout/index.tsx file.